src/App.tsx

```diff
@@ -1,61 +1,43 @@
+import { useState } from 'react';
+
 import './App.scss';
+import todosFromServer from './api/todos';
+
+import { getUserById } from './services/user';
+import { TodosWithUser } from './types/TodosWithUser';
+
+import { TodoForm } from './components/TodoForm/TodoForm';
+import { TodoList } from './components/TodoList';
+
+export const todos: TodosWithUser[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getNewTodoId(todosArray: TodosWithUser[]) {
+  const maxId = Math.max(...todosArray.map(todo => todo.id));
+
+  return maxId + 1;
+}
 
 export const App = () => {
+  const [todo, setTodo] = useState<TodosWithUser[]>(todos);
+
+  const addTodo = ({ id, ...data }: TodosWithUser) => {
+    const newTodo = {
+      id: getNewTodoId(todo),
+      ...data,
+    };
+
+    setTodo(currentTodo => [...currentTodo, newTodo]);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+      <TodoForm onSubmit={addTodo} />
```

Comment:
Ensure that the `TodoForm` component correctly handles the form submission logic using the `onSubmit` prop, as specified in the checklist .